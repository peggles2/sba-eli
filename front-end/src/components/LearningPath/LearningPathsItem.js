import React, { Component } from 'react'
import { Card, Image } from "semantic-ui-react";
import LearningPathProgress from "./LearningPathProgress";
import axios from "axios";
import { withRouter } from 'react-router-dom';
import { connect } from "react-redux";

import { getLearningEvents } from "../../actions/learningEventActions";

export class LearningPathsItem extends Component {
  state = { topicsList: [] }

  componentDidMount() {
    this.getTopicsList();
  }

  learningPathProgress = (total) => {
    // const isUserLoggedIn = this.props.isUserLoggedIn
    const isUserLoggedIn = true;
    if(isUserLoggedIn) {
      const complete = 0;

      return (
        <Card.Content>
          <LearningPathProgress complete={complete} total={total}/>
        </Card.Content>
      );
    }
  };

  getTopicsList = () => {
    const baseUrl = process.env.REACT_APP_SERVICE_HOST;

    axios.get(baseUrl + `/learning_objectives/`, {params: {course_id: this.props.id}})
      .then(res => {
        const topicsList = res.data
        this.setState({ topicsList })
      })
  }

  totalMicroLearningEvents = () => {
    const reducer = (totalList, number) => totalList + number;
    const topicsList = this.state.topicsList;

    if(topicsList.length > 0) {
      const total = topicsList.map(topic => topic.items_count).reduce(reducer);

      return total;
    } else {
      return 0
    }
  }

  render() {
    const total = this.totalMicroLearningEvents();
    const learningPathDescription = "Maybe it means something more - something we can't yet understand."

    return(
      <Card>
        <Image
          src={`/Image_Placeholder.png`}
          alt="learning path image placeholder"
        />
        <Card.Content>
          <Card.Header aria-label={`Link to Learning Path ` + this.props.name}>
            <a href="#" onClick={() => { this.props.history.push('/learning_paths/' + this.props.id)}}>{this.props.name}</a>
          </Card.Header>
          <Card.Meta>{this.props.course_code}</Card.Meta>
          <Card.Description>{learningPathDescription}</Card.Description>
        </Card.Content>
        {this.learningPathProgress(total)}
      </Card>
    )
  }
};

const mapStateToProps = (store) => {
  return {
    isUserLoggedIn: store.login.isUserLoggedIn,
  }
}

export default withRouter(connect(mapStateToProps)(LearningPathsItem));
