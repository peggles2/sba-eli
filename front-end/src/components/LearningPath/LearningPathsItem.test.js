import React from 'react';
import LearningPathsItem from './LearningPathsItem';
import { shallow, mount } from 'enzyme';

describe('LearningPath', () => {
  it('should render correctly', () => {
    const item = { id: 1, name: "LP 1" }
    const wrapper = shallow(<LearningPathsItem item={item} />); 
    
    expect(wrapper).toMatchSnapshot();
  });
  
  it('should render a <li>', () => {
    const item = { id: 1, name: "LP 1" }
    const wrapper = shallow(<LearningPathsItem item={item} />); 

    expect(wrapper.find('li').length).toEqual(1);
  });
});
