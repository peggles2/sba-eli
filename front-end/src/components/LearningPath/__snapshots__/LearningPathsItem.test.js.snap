// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LearningPath should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LearningPathsItem
    item={
      Object {
        "id": 1,
        "name": "LP 1",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Learning Path: ",
        "LP 1",
      ],
    },
    "ref": null,
    "rendered": Array [
      "Learning Path: ",
      "LP 1",
    ],
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Learning Path: ",
          "LP 1",
        ],
      },
      "ref": null,
      "rendered": Array [
        "Learning Path: ",
        "LP 1",
      ],
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
